unstructured_ingest/v2/processes/connectors/discord.py

```diff
@@ -0,0 +1,156 @@
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+    def run(self, **kwargs: Any) -> Generator[FileData, None, None]:
+        client = self.get_client()
+        channels_to_process: set[str] = set(self.connection_config.channels or [])
+
+        for channel_id in list(channels_to_process):
+            file_data = self.get_channel_file_data(channel_id=channel_id, client=client)
+            if file_data:
+                yield file_data
+
+    @requires_dependencies(["discord"], extras="discord")
+    def get_channel_file_data(self, channel_id: str, client) -> Optional[FileData]:
+        # Fetch channel metadata
+        dt.datetime.utcnow().isoformat()
+        identifier = channel_id
+        channel_id = f"{channel_id}.txt"
+        source_identifiers = SourceIdentifiers(
+            filename=identifier,
+            fullpath=channel_id,
+            rel_path=identifier,
+        )
+        metadata = FileDataSourceMetadata(
+            record_locator={"channel_id": identifier},
+            date_processed=str(dt.datetime.utcnow().isoformat()),
+        )
+        return FileData(
+            identifier=identifier,
+            connector_type=CONNECTOR_TYPE,
+            source_identifiers=source_identifiers,
+            metadata=metadata,
+        )
+
+
+class DiscordDownloaderConfig(DownloaderConfig):
+    @requires_dependencies(["discord"], extras="discord")
+    def get_client(self):
+        import discord
+
+        intents = discord.Intents.default()
+        intents.message_content = True
+        return discord.Client(intents=intents)
+
+
+@dataclass
+class DiscordDownloader(Downloader):
+    connection_config: DiscordConnectionConfig
+    download_config: DiscordDownloaderConfig
+
+    @requires_dependencies(["discord"], extras="discord")
+    def load_async(self):
+        import discord
+
+        return discord.Client, None
+
+    def run(self, file_data: FileData, **kwargs: Any) -> DownloadResponse:
+        # Synchronous run is not implemented
+        raise NotImplementedError()
+
+    async def run_async(self, file_data: FileData, **kwargs: Any) -> DownloadResponse:
+        import discord
+        from discord.ext import commands
+
+        client = self.download_config.get_client()
+        record_locator = file_data.metadata.record_locator
+
+        if "channel_id" in record_locator:
+            bot = commands.Bot(command_prefix=">", intents=client.intents)
+            download_path = self.get_download_path(file_data=file_data)
+            download_path.parent.mkdir(parents=True, exist_ok=True)
+
+            messages: list[discord.Message] = []
+
+            @bot.event
+            async def on_ready():
+                logging.info("Bot is ready")
+                record_locator["channel_id"] = record_locator["channel_id"].split(".")[0]
```

Comment:
What does this update do? And if we need it, it should be done when the channel id is set on the file data in the indexer. Mind posting an example of when the channel id has a period in it and we need to split on it?

Reply:
This was never answered?